superset-frontend/plugins/plugin-chart-period-over-period-kpi/src/plugin/transformProps.ts

```diff
@@ -0,0 +1,129 @@
+  /**
+   * This function is called after a successful response has been
+   * received from the chart data endpoint, and is used to transform
+   * the incoming data prior to being sent to the Visualization.
+   *
+   * The transformProps function is also quite useful to return
+   * additional/modified props to your data viz component. The formData
+   * can also be accessed from your CustomViz.tsx file, but
+   * doing supplying custom props here is often handy for integrating third
+   * party libraries that rely on specific props.
+   *
+   * A description of properties in `chartProps`:
+   * - `height`, `width`: the height/width of the DOM element in which
+   *   the chart is located
+   * - `formData`: the chart data request payload that was sent to the
+   *   backend.
+   * - `queriesData`: the chart data response payload that was received
+   *   from the backend. Some notable properties of `queriesData`:
+   *   - `data`: an array with data, each row with an object mapping
+   *     the column/alias to its value. Example:
+   *     `[{ col1: 'abc', metric1: 10 }, { col1: 'xyz', metric1: 20 }]`
+   *   - `rowcount`: the number of rows in `data`
+   *   - `query`: the query that was issued.
+   *
+   * Please note: the transformProps function gets cached when the
+   * application loads. When making changes to the `transformProps`
+   * function during development with hot reloading, changes won't
+   * be seen until restarting the development server.
+   */
+  const {
+    width,
+    height,
+    formData,
+    queriesData,
+    datasource: { currencyFormats = {}, columnFormats = {} },
+  } = chartProps;
+  const {
+    boldText,
+    headerFontSize,
+    headerText,
+    metrics,
+    yAxisFormat,
+    currencyFormat,
+    subheaderFontSize,
+  } = formData;
+  const dataA = queriesData[0].data as TimeseriesDataRecord[];
+  const dataB = queriesData[1].data as TimeseriesDataRecord[];
+  const data = dataA;
+  const metricName = getMetricLabel(metrics[0]);
+  let bigNumber = data.length === 0 ? null : data[0][metricName];
+  let prevNumber = dataB.length === 0 ? null : dataB[0][metricName];
+
+  const numberFormatter = getValueFormatter(
+    metrics[0],
+    currencyFormats,
+    columnFormats,
+    yAxisFormat,
+    currencyFormat,
+  );
+
+  const compTitles = {
+    r: 'Range' as string,
+    y: 'Year' as string,
+    m: 'Month' as string,
+    w: 'Week' as string,
+  };
+
+  const formatPercentChange = getNumberFormatter(
+    NumberFormats.PERCENT_SIGNED_1_POINT,
+  );
+
+  let valueDifference = bigNumber - prevNumber;
+
+  const percentDifferenceNum = prevNumber
+    ? (bigNumber - prevNumber) / Math.abs(prevNumber)
+    : 0;
+
+  const compType= compTitles[formData.timeComparison]
+  bigNumber = numberFormatter(bigNumber);
+  prevNumber = numberFormatter(prevNumber);
+  valueDifference = numberFormatter(valueDifference);
+  const percentDifference: string = formatPercentChange(percentDifferenceNum);
+
+  return {
+    width,
```

Comment:
We should only return those values that are used in the main plugin file. I see that don't use for example data, metrics or metricName in `PopKPI` so we should remove those from here